server/container_create.go

```diff
@@ -737,6 +737,11 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 		}
 	}
 
+	// Only allow strace annotation if cri-o daemon allows it
+	if !s.config.Config.AllowStrace {
+		specgen.RemoveAnnotation(annotations.Strace)
```

Comment:
I'm not sure if the server config is available in `CreateContainer` or not.  But this approach only covers annotations and does not guard against the setting coming in through labels.  It would be nice if we could move this guard into `CreateContainer` somehow to cover both injection paths.  As a side benefit, it would allow us to guard against unwanted `strace` invocations without having to adjust user-supplied annotations (more on leaving annotations alone in opencontainers/runtime-spec#946).